include/envoy/server/listener_manager.h

```diff
@@ -131,6 +131,14 @@ class ListenerManager {
     All,
   };
 
+  enum class ListenerState {
+    ACTIVE = 1,
+    WARMING = 2,
+    DRAINING = 3,
+    // all active and warming listeners, but not draining
+    ALL = 4
```

Comment:
`ALL` should include all of the listeners.  Most readers of code which uses the enum will not see the comment here.

How about making each enum value set a bit:

```
enum class ListenerState {
    ACTIVE = 1<<0,
    WARMING = 1<<1,
    DRAINING = 1<<2,
}
```

Then you can choose which states you want by ORing the desired states together:

```
listener_manager_.listeners(ListenerState::ACTIVE|ListenerState::WARMING));
```

In the implementation of `listeners(ListenerState state)`, you can see if an enum should be included like so:

```
if (state & ACTIVE) {
  // Add active listeners.
}
if (state & WARMING) {
  // Add warming listeners.
}
```

Reply:
Good idea thanks!